crates/circuit/src/dag_circuit.rs

```diff
@@ -434,40 +415,64 @@ struct DAGVarInfo {
     out_node: NodeIndex,
 }
 
+#[derive(Clone, Debug)]
+struct VarsByType {
+    input: OnceLock<Py<PySet>>,
+    capture: OnceLock<Py<PySet>>,
+    declare: OnceLock<Py<PySet>>,
+}
+
+impl<'py> VarsByType {
+    fn new() -> Self {
+        Self {
+            input: OnceLock::new(),
+            capture: OnceLock::new(),
+            declare: OnceLock::new(),
+        }
+    }
+
+    fn get_input_bound(&'py self, py: Python<'py>) -> &'py Bound<'py, PySet> {
```

Comment:
I think these names are no longer hip?

```suggestion
    fn get_input(&'py self, py: Python<'py>) -> &'py Bound<'py, PySet> {
```

Reply:
Hah, nobody has ever accused me of being hip. But I can drop the `_bound` suffixes everywhere.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/14069/commits/1f56ec5bf850a5cf459d8a9465febc40f9603a99